### PR TITLE
Update tables for better use of API filters, context cancellation in list calls and page limiting for limit clause in query closes #688

### LINKED ISSUE #688: Update tables for better use of API filters, context cancellation in list calls and page limiting for limit clause in query
**Is your feature request related to a problem? Please describe.**
Check and address all tables except [committed ones](https://github.com/turbot/steampipe-plugin-aws/commit/c09346ae80a8bcb1d58c6827e64f6fed501a1aab)  (reference).

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ParthaI**
- aws_accessanalyzer_analyzer
- aws_acm_certificate
- aws_api_gateway_api_authorizer
- aws_api_gateway_api_key
- aws_api_gateway_rest_api
- aws_api_gateway_stage
- aws_api_gateway_usage_plan
- aws_api_gatewayv2_api
- aws_appautoscaling_target
- aws_auditmanager_assessment
- aws_auditmanager_evidence
- aws_auditmanager_evidence_folder
- aws_availability_zone
- aws_backup_plan
- aws_backup_protected_resource
- aws_backup_recovery_point
- aws_backup_selection
- aws_cloudcontrol_resource
- aws_cloudformation_stack
- aws_cloudfront_cache_policy
- aws_cloudfront_distribution
- aws_cloudfront_origin_access_identity
- aws_cloudfront_origin_request_policy
- aws_cloudwatch_alarm
- aws_cloudwatch_log_group
- aws_cloudwatch_log_metric_filter
- aws_cloudwatch_log_stream
- aws_codebuild_project
- aws_codebuild_source_credential
- aws_codecommit_repository
- aws_codepipeline_pipeline
- aws_config_configuration_recorder
- aws_config_conformance_pack
- aws_config_rule
- aws_dax_cluster
- aws_directory_service_directory
- aws_dms_replication_instance
- aws_dynamodb_backup
- aws_dynamodb_global_table
- aws_dynamodb_table
- aws_ebs_snapshot

### CODE REVIEW

**Comment by cbruno10** [comment]
@ParthaI Please address failing checks in the golangci-lint workflow run

**Comment by ParthaI** [comment]
> @ParthaI Please address failing checks in the golangci-lint workflow run

Sure @cbruno10 will check this one I have already noticed it.

File: aws/table_aws_config_rule.go [hunk 1]
```diff
@@ -121,11 +127,24 @@ func listConfigRules(ctx context.Context, d *plugin.QueryData, _ *plugin.Hydrate
 		return nil, err
 	}
 
+	input := &configservice.DescribeConfigRulesInput{}
+
+	// Additonal Filter
+	equalQuals := d.KeyColumnQuals
+	if equalQuals["name"] != nil {
+		input.ConfigRuleNames = []*string{aws.String(equalQuals["type"].GetStringValue())}
```
Comment (bigdatasourav) [inline aws/table_aws_config_rule.go:135]:
```suggestion
		input.ConfigRuleNames = []*string{aws.String(equalQuals["name"].GetStringValue())}
```

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Pull request merged**